cpp/example_code/rds/getting_started_with_db_instances.cpp

```diff
@@ -0,0 +1,1079 @@
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            ++counter;
+            if (counter > 600) {
+                std::cerr << "Wait for snapshot to be available timed out ofter "
+                          << counter
+                          << " seconds." << std::endl;
+                cleanUpResources(PARAMETER_GROUP_NAME, DB_INSTANCE_IDENTIFIER, client);
+                return false;
+            }
+
+            // 14. Wait for the snapshot to become available.
+            // snippet-start:[cpp.example_code.rds.DescribeDBSnapshots]
+            Aws::RDS::Model::DescribeDBSnapshotsRequest request;
+            request.SetDBSnapshotIdentifier(snapshotID);
+
+            Aws::RDS::Model::DescribeDBSnapshotsOutcome outcome =
+                    client.DescribeDBSnapshots(request);
+
+            if (outcome.IsSuccess()) {
+                snapshot = outcome.GetResult().GetDBSnapshots()[0];
+            }
+            else {
+                std::cerr << "Error with RDS::DescribeDBSnapshots. "
+                          << outcome.GetError().GetMessage()
+                          << std::endl;
+                cleanUpResources(PARAMETER_GROUP_NAME, DB_INSTANCE_IDENTIFIER, client);
+                return false;
+            }
+            // snippet-end:[cpp.example_code.rds.DescribeDBSnapshots]
+
+            if ((counter % 20) == 0) {
+                std::cout << "Current snapshot status is '"
+                          << snapshot.GetStatus()
+                          << "' after " << counter << " seconds." << std::endl;
+            }
+        } while (snapshot.GetStatus() != "available");
+
+        if (snapshot.GetStatus() != "available") {
+            std::cout << "A snapshot has been created." << std::endl;
+        }
+    }
+
+    printAsterisksLine();
+
+    bool result = true;
+    if (askYesNoQuestion(
+            "Do you want to delete the DB instance and parameter group (y/n)? ")) {
+        result = cleanUpResources(PARAMETER_GROUP_NAME, DB_INSTANCE_IDENTIFIER, client);
+    }
+
+    return result;
+}
+
+//! Routine which gets DB parameters using the 'DescribeDBParameters' api.
+/*!
+ \sa getDBParameters()
+ \param parameterGroupName: The name of the parameter group.
+ \param namePrefix: Prefix string to filter results by parameter name.
+ \param source: A source such as 'user', ignored if empty.
+ \param parametersResult: Vector of 'Parameter' objects returned by the routine.
+ \param client: 'RDSClient' instance.
+ \return bool: Successful completion.
+ */
+bool AwsDoc::RDS::getDBParameters(const Aws::String &parameterGroupName,
+                                  const Aws::String &namePrefix,
+                                  const Aws::String &source,
+                                  Aws::Vector<Aws::RDS::Model::Parameter> &parametersResult,
+                                  const Aws::RDS::RDSClient &client) {
+    // snippet-start:[cpp.example_code.rds.DescribeDBParameters]
```

Comment:
This is optional, but I think these comment blocks above the snippet tag are quite useful in establishing the context of the action call. Have you considered including them so they are shown in the docs? I'm only commenting on it here, but same for all of the actions. I did not fully understand them in the docs without this comment block and declaration. It may not work well because you are including things across methods, but something to think about.

Reply:
Good suggestion.